src/main/io/vtx.c

```diff
@@ -80,30 +82,40 @@ void vtxProcess(timeUs_t currentTimeUs)
         if (currentTimeUs > lastCycleTimeUs + VTX_PARAM_CYCLE_TIME_US) {
             switch (currentSchedule) {
             case VTX_PARAM_BANDCHAN:
-                if (vtxSettingsConfig()->band) {
-                    uint8_t vtxBand;
-                    uint8_t vtxChan;
-                    if (vtxCommonGetBandAndChannel(&vtxBand, &vtxChan)) {
-                        if (vtxSettingsConfig()->band != vtxBand || vtxSettingsConfig()->channel != vtxChan) {
-                            vtxCommonSetBandAndChannel(vtxSettingsConfig()->band, vtxSettingsConfig()->channel);
+                if(!ARMING_FLAG(ARMED)) {
```

Comment:
Wouldn't it be safer to not call vtxCommonProcess() when armed and power has been switched? If e.g. vtxCommonSetBandAndChannel() gets called from anywhere else (due to a programmer bug), not calling vtxCommonProcess() would at least ensure you don't kill the video while flying.

Reply:
well it will need to be called until eventually consistent.. no way to do that until consistency is reached.  maybe a static boolean check while armed?

Reply:
due to the async nature there's no guarantee that it will change on the first shot.

Reply:
Maybe you can have a quick look at https://github.com/lostcontrol/betaflight/commit/11724510019b512afd798092a487a06857336a1f in src/main/drivers/vtx_common.c. It was based on a previous state of your PR. If you see anything worth taking, feel free 😉

Reply:
Added a state check below that should do exactly what it is you're asking.